packages/core/src/test/awsService/appBuilder/serverlessLand/main.test.ts

```diff
@@ -114,16 +118,51 @@ describe('downloadPatternCode', () => {
         assert(getPatternStub.firstCall.args[4] === true)
     })
     it('handles download failure', async () => {
-        const mockAssetName = 'test-project-sam-python.zip'
         const error = new Error('Download failed')
         getPatternStub.rejects(error)
         try {
-            await downloadPatternCode(mockConfig, mockAssetName)
+            await downloadPatternCode(mockConfig, mockConfig.assetName)
             assert.fail('Expected an error to be thrown')
         } catch (err: any) {
             assert.strictEqual(err.message, 'Failed to download pattern: Error: Download failed')
         }
     })
+    it('downloads pattern when directory exists and user confirms overwrite', async function () {
```

Comment:
This is a huge amount of test code (and brittle, because of the over-use of mocks/stubs), just to test a  function that checks for a file before overwriting it.

And this is another very strong example of why using shared code is valuable: if this code was generalized, we wouldn't need this new test code at all. 

Every batch of 30+ lines of redundant code is almost entirely a cost, for very small benefit. So it's important to think about re-using code, especially for cases like this where there is a clear path to doing so. Look at messages.ts  or other modules to see if there's existing shared util that does this check-before-save-and-show-message logic. It's really common, and if it doesn't already exist, then this is a good time to do it.